packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -918,9 +914,9 @@ class ServiceExtensionManager {
       await _callServiceExtension(name, value);
     }
 
-    final StreamController<ServiceExtensionState> streamController =
-        _getServiceExtensionStateController(name);
-    streamController.add(ServiceExtensionState(enabled, value));
+    // TODO(jacobr): we might want to be a little smarter about when the value
+    // hasn't really changed.
+    _serviceExtensionState(name).value = ServiceExtensionState(enabled, value);
```

Comment:
I believe that ValueNotifier is smart about this and only notifies listeners if the value is changed. As long as ServiceExtensionState implements equality, this should be taken care of automatically.

Reply:
Added equals and hashcode to ServiceExtensionState and removed todo.